lib/ts/Diags.cc

```diff
@@ -625,11 +625,14 @@ Diags::should_roll_diagslog()
         fflush(diags_log->m_fp);
         if (diags_log->roll()) {
```

Comment:
The old file  don't need to  be closed in  diags_log->roll(). close old file should at destructor of old_diags.  in addition,
this will cause crash because of writing to closed file